[TGUI] Smartfridge

RedWolfie opened the pull request:
## What Does This PR Do
Adds TGUI for smartfridge

## Why It's Good For The Game
Gets rid of Nano

## Images of changes
![smartfridge-demo2](https://user-images.githubusercontent.com/13683223/93343050-56680e00-f830-11ea-93d0-2360a5ca736a.png)

Empty
![smartfridge-demo1](https://user-images.githubusercontent.com/13683223/93343052-56680e00-f830-11ea-887c-21fde711822b.png)

Secure variant
![smartfridge-demo3](https://user-images.githubusercontent.com/13683223/93343048-55cf7780-f830-11ea-8880-c389d43e0942.png)

Drying rack variant
![smartfridge-demo4](https://user-images.githubusercontent.com/13683223/93343047-5536e100-f830-11ea-8e97-af7fc5f081ce.png)

Old one for reference
![fridge-ref1](https://user-images.githubusercontent.com/13683223/93343277-9a5b1300-f830-11ea-90be-d07a5e5a7778.png)


## Changelog
:cl:
add: Smartfridge TGUI
/:cl:

AffectedArc07 commented on code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:
```diff
@@ -139,15 +139,13 @@
 
 /obj/machinery/smartfridge/attackby(obj/item/O, var/mob/user)
 	if(exchange_parts(user, O))
-		SSnanoui.update_uis(src)
```
Put this back with `SStgui.update_uis(src)`

AffectedArc07 commented on code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:
```diff
@@ -139,15 +139,13 @@
 
 	if(load(O, user))
 		user.visible_message("<span class='notice'>[user] has added \the [O] to \the [src].</span>", "<span class='notice'>You add \the [O] to \the [src].</span>")
-		SSnanoui.update_uis(src)
```
Same here

AffectedArc07 commented on code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:
```diff
@@ -157,7 +155,6 @@
 				items_loaded++
 		if(items_loaded)
 			user.visible_message("<span class='notice'>[user] loads \the [src] with \the [P].</span>", "<span class='notice'>You load \the [src] with \the [P].</span>")
-			SSnanoui.update_uis(src)
```
And here

AffectedArc07 commented on code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:
```diff
@@ -380,7 +369,6 @@
 	if(href_list["vend"] && (usr.contents.Find(src) || Adjacent(usr)))
 		if(!emagged && locked != SMART_FRIDGE_LOCK_SHORTED && scan_id && !allowed(usr))
 			to_chat(usr, "<span class='warning'>Access denied.</span>")
-			SSnanoui.update_uis(src)
```
You need to update the `/Topic` call here, since it no longer used with TGUI

AffectedArc07 commented on code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:
```diff
@@ -741,14 +720,12 @@
 			new dried(loc)
 			item_quants[S.name]--
 			qdel(S)
-		SSnanoui.update_uis(src)
 		return TRUE
 	for(var/obj/item/stack/sheet/wetleather/WL in contents)
 		var/obj/item/stack/sheet/leather/L = new(loc)
 		L.amount = WL.amount
 		item_quants[WL.name]--
 		qdel(WL)
-		SSnanoui.update_uis(src)
```
Update these to `SStgui.update_uis(src)`

AffectedArc07 commented on code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:
```diff
@@ -373,14 +365,13 @@
 
-/obj/machinery/smartfridge/secure/Topic(href, href_list)
+/obj/machinery/smartfridge/secure/tgui_act(action, params)
 	if(stat & (BROKEN|NOPOWER))
```
Here it *is* possible to use the buttons here since the `..()` check isnt at the start. Someone *could* be able to attempt to eject an item from range, but the parent call at the bottom will stop it. So they can be __denied__ at range, but not granted.

Personally I would take the check below and just bolt it into the regular smartfridge `tgui_act`, while checking for `is_secure` in this proc, but this *should* be fine. Up to you really.

RedWolfie replied:
done

RedWolfie commented:
Changes
Moved the secure check into the regular fridge
Updated the warning in the secure fridge to use Noticebox instead of Box
Deleted the locked var as it did the same as the emagged variable
Deleted some unused variables in data

![sec](https://user-images.githubusercontent.com/13683223/93620829-e649bb80-f9da-11ea-8f92-e7191d813e94.png)
